src/components/filters/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, ReactNode } from "react"
+
+import { formatNumber } from "../../lib/formattingHelper"
+
+import Checkbox from "../../components/checkbox"
+
+interface Item {
+  value: string | number
+  label: string
+  renderIcon?: () => ReactNode
+}
+
+interface Props {
+  name: string
+  options: Item[]
+  applyFilters: (filtersToApply) => void
+  selected?: (string | number)[]
+  facet?: { [key: string]: number }
+}
+
+const CheckboxFilter: FC<Props> = ({
+  name,
+  options,
+  applyFilters,
+  selected = [],
+  facet = {},
+  children,
+}) => {
+  const values = options.map(({ value }) => value)
+
+  const selectValue = (value) => () => {
+    const selectionCopy = [...selected]
+
+    if (selectionCopy.includes(value)) {
+      const i = selectionCopy.indexOf(value)
+      selectionCopy.splice(i, 1)
+    } else {
+      selectionCopy.push(value)
+    }
+
+    selectionCopy.sort((a, b) =>
+      values.indexOf(a) < values.indexOf(b) ? -1 : 1
+    )
+
+    applyFilters({ [name]: selectionCopy })
+  }
+
+  const renderOption = ({ value, label, renderIcon }: Item) => {
+    const isSelected = selected.includes(value)
+    const resultsCount = facet[value.toString()]
+    const isDisabled = resultsCount === 0 && !isSelected
+
+    return (
+      <div className="w-12/12 -mb-5" key={`filter_${name}_${value}`}>
```

Comment:
What is the negative margin used for?

Reply:
To reduce the space that comes from the checkbox.

Reply:
Isn't that breaking the space needed to display a potential validation error then?

Reply:
It could if single checkbox could ever show the validation error. That is not the case though.